src/nvim/api/buffer.c

```diff
@@ -291,7 +291,11 @@ void buffer_set_line_slice(Buffer buffer,
   // Only adjust marks if we managed to switch to a window that holds
   // the buffer, otherwise line numbers will be invalid.
   if (save_curbuf == NULL) {
-    mark_adjust((linenr_T)start, (linenr_T)(end - 1), MAXLNUM, extra);
+    /// WCONV: extra is ssize_t, representing lines added or removed, so
+    /// WCONF: logically it should be < MAXLNUM, which is INT32_MAX
+    /// WCONF: Add an assertion to be sure
+    assert( -MAXLNUM <= extra && extra <= MAXLNUM);
```

Comment:
I don't think `linenr_T` is right, since `extra` designates a number of lines (as in a quantity) rather than line numbers (as in addresses). As such, I think the type of `extra` is determined by the type of `linenr_T`. `ptrdiff_t` sounds reasonable, but might be to big (probably not an issue).